Content.Server/Clothing/Components/TransformableClothingComponent.cs

```diff
@@ -0,0 +1,31 @@
+using Robust.Shared.Audio;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.Clothing.Components;
+
+
+/// <summary>
+/// This component is used for clothing that can be transformed entirely
+/// from one form to another, e.g. head bandanas to face bandanas
+/// </summary>
+[RegisterComponent]
+public sealed partial class TransformableClothingComponent : Component
+{
+    /// <summary>
+    /// The prototype this clothing will be turned into.
+    /// </summary>
+    [DataField(required: true)]
+    public EntProtoId TransformProto;
+
+    /// <summary>
+    /// How long it takes to transform the clothing.
+    /// </summary>
+    [DataField]
+    public TimeSpan TransformDelay = TimeSpan.FromSeconds(1.5f);
+
+    /// <summary>
+    /// The sound to play when transforming the clothing.
+    /// </summary>
+    [DataField]
+    public SoundSpecifier? TransformSound = new SoundCollectionSpecifier("storageRustle", new AudioParams().WithVolume(6));
+}
```

Comment:
isnt this basically just duplicating `FoldingSystem`? i.e. altverb to transform an entity from one to another, i dont see any reason this should be clothing specific

Reply:
FoldableSystem doesn't transform an entity into another, it just updates the appearance and toggles some functionality. The folded version entities I think are just there for mapping/box fills while starting in folded=true state.

I'll see if I can make this work with Foldable system. Masks only have a few components to toggle, so it might be doable.